src/clients/c++/request.h

```diff
@@ -666,6 +667,29 @@ class InferContext {
   Error GetReadyAsyncRequest(
       std::shared_ptr<Request>* async_request, bool wait);
 
+  /// Send current request through the context's stream. All requests sent will
+  /// be directed to the same model instance, be processed, and be returned
+  /// in the same order. Note that you need to retrieve the responses with
+  /// GetStreamRunResults() method.
+  ///
+  /// \return Error object indicating success or failure.
+  virtual Error StreamRun() = 0;
```

Comment:
Having a separate API for this doesn't seem right. Instead it seems like it should just be an option to Run and RunAsync. You can do either synchronous or asynchronous over a stream, right?

Reply:
Actually I think the stream / no-stream should be on the InferContext.

Reply:
The gRPC bidirectional streaming already have the decoupling between sending requests and receiving responses. So even the "sync streaming", it is similar to async infer in some sense, after sending a request, the client can perform other operations before receiving the response.